src/runtime/d3d12compute.cpp

```diff
@@ -3023,8 +3005,9 @@ WEAK int halide_d3d12compute_run(void *user_context,
     StartCapturingGPUActivity();
 #endif
 
-    halide_assert(user_context, state_ptr);
-    module_state *state = (module_state *)state_ptr;
+    d3d12_library *library;
```

Comment:
Does `lookup()` guarantee null-init on failure? If not, init to null here seems prudent.